src/main/java/core/basesyntax/JavaDateTimeApi.java

```diff
@@ -1,125 +1,178 @@
+        return localDate.plusWeeks(numberOfWeeks);
     }
 
     /**
      * Given a random `someDate` date, return one of the following Strings:
-     *     - "`someDate` is after `currentDate`"
-     *                  if `someDate` is in the future relating to the `current date`;
-     *     - "`someDate` is before `currentDate`"
-     *                  if `someDate` is in the past relating to the `current date`;
-     *     - "`someDate` is today"
-     *                  if `someDate` is today;
+     * - "`someDate` is after `currentDate`"
+     * if `someDate` is in the future relating to the `current date`;
+     * - "`someDate` is before `currentDate`"
+     * if `someDate` is in the past relating to the `current date`;
+     * - "`someDate` is today"
+     * if `someDate` is today;
      */
     public String beforeOrAfter(LocalDate someDate) {
-        return someDate + "is today";
+        LocalDate current = LocalDate.now();
+        return someDate.isBefore(current) ? someDate + " is before " + current
+                : someDate.isAfter(current) ? someDate + " is after " + current
+                : someDate + " is today";
     }
 
     /**
      * Given a String representation of a date and some timezone,
      * return LocalDateTime in this timezone.
      */
     public LocalDateTime getDateInSpecificTimeZone(String dateInString, String zone) {
-        return LocalDateTime.now();
+        return LocalDateTime.from(ZonedDateTime.parse(dateInString)
+                .withZoneSameInstant(ZoneId.of(zone)));
     }
 
     /**
      * Given some LocalDateTime, return an OffsetDateTime with the local time offset applied
      * (`+02:00` for Ukraine).
-     *
+     * <p>
      * Example: we receive a LocalDateTime with a value `2019-09-06T13:17`.
-     *          We should return the OffsetDateTime with a value `2019-09-06T13:17+02:00`,
-     *          where `+02:00` is the offset for our local timezone.
-     *
+     * We should return the OffsetDateTime with a value `2019-09-06T13:17+02:00`,
+     * where `+02:00` is the offset for our local timezone.
+     * <p>
      * OffsetDateTime is recommended to use for storing date values in a database.
      */
     public OffsetDateTime offsetDateTime(LocalDateTime localTime) {
-        return OffsetDateTime.now();
+        ZoneId ukraineZoneId = ZoneId.of(ZoneId.SHORT_IDS.get(SHORT_IDS_ECT));
+        ZoneOffset zoneOffset = ukraineZoneId.getRules().getOffset(localTime);
+        return OffsetDateTime.of(localTime, zoneOffset);
     }
 
     /**
      * Given a String formatted as `yyyymmdd`,
      * return Optional of this date as a LocalDate.
      */
     public Optional<LocalDate> parseDate(String date) {
-        return Optional.empty();
+        DateTimeFormatter dateTimeFormatter =
+                DateTimeFormatter.ofPattern(PATTERN_DATE_YYYY_MM_DD);
+        try {
+            return Optional.ofNullable(LocalDate.parse(date, dateTimeFormatter));
```

Comment:
Optional.ofNullable() can still produce the same result as Optional.empty(); Let's use Optional.of() here

Reply:
done